### PR TITLE
soc: Remove superfluous kbuild help text

### PR DESCRIPTION
**Pull request by nordicjm**
Zephyr moved away from kbuild in 2017, the help text for SOC and SOC_SERIES Kconfigs was outdated at that time but was never removed, remove instances where the help text (which is now completely hidden from the user) offers nothing over the Kconfig symbol name itself

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by nordicjm** [comment]
> how is a help to a kconfig is related to kbuild in this case? I am missing the connection.

Because these were selected and visible when kbuild was used, in hwmv1 there were also selectable but changing them would break the build, in hwmv2 these are hidden

File: soc/arm/musca/Kconfig.soc [hunk 1]
```diff
@@ -4,20 +4,14 @@
...
 	bool
 	select SOC_FAMILY_ARM
-	help
-	   ARM v2m MUSCA MCU Series
 
 config SOC_V2M_MUSCA_B1
 	bool
 	select SOC_SERIES_MUSCA
-	help
-	  ARM Cortex-M33 SMM-SSE-200 on V2M-MUSCA-B1
```
Comment (stephanosio) [inline soc/arm/musca/Kconfig.soc:14]: The description provided here looks pretty useful to me -- just by looking at `SOC_V2M_MUSCA_B1`, it is not very easy to figure out that this is for SSE-200 reference system.

File: soc/atmel/sam/sam4l/Kconfig.soc [hunk 2]
```diff
@@ -4,12 +4,6 @@
 config SOC_SERIES_SAM4L
 	bool
 	select SOC_FAMILY_ATMEL_SAM
-	help
-	  Enable support for Atmel SAM4L Cortex-M4 microcontrollers.
-	  Part No.: SAM4LS8C, SAM4LS8B, SAM4LS8A, SAM4LS4C, SAM4LS4B,
-	  SAM4LS4A, SAM4LS2C, SAM4LS2B, SAM4LS2A, SAM4LC8C, SAM4LC8B,
-	  SAM4LC8A, SAM4LC4C, SAM4LC4B, SAM4LC4A SAM4LC2C, SAM4LC2B,
-	  SAM4LC2A
```
Comment (stephanosio) [inline soc/atmel/sam/sam4l/Kconfig.soc:12]: could be very useful when grepping for part number (e.g. what `SOC_SERIES` do I need to select for `SAM4LC2C`?)

File: soc/xlnx/zynq7000/xc7zxxxs/Kconfig.soc [hunk 3]
```diff
@@ -9,9 +9,6 @@
 config SOC_SERIES_XC7ZXXXS
 	bool
 	select SOC_FAMILY_XILINX_ZYNQ7000
-	help
-	  Enable support for the Xilinx Zynq-7000S (XC7ZxxxS)
-	  SoC series (single core ARM Cortex-A9).
```
Comment (stephanosio) [inline soc/xlnx/zynq7000/xc7zxxxs/Kconfig.soc:14]: also looks pretty useful to me -- `SOC_SERIES_XC7ZXXXS` is quite cryptic.

**Review by stephanosio — COMMENTED** [review]
> in hwmv2 these are hidden

Even if these are hidden/prompt-less, the description provided under "help" can still be useful while grepping or using the ["Kconfig search"](https://docs.zephyrproject.org/latest/kconfig.html).

**Comment by nashif** [comment]
> Even if these are hidden/prompt-less, the description provided under "help" can still be useful while grepping or using the ["Kconfig search"](https://docs.zephyrproject.org/latest/kconfig.html).

yeah, I wonder if the presence of the help is causing any issues somewhere. Verbosity in the kconfig and knowing what each option is about while looking at kconfig files is very useful, even if it is not going to be displayed in tools.

File: soc/atmel/sam/sam4l/Kconfig.soc (hunk 2 above)
Comment (nordicjm) [reply soc/atmel/sam/sam4l/Kconfig.soc:12]: You don't select a SOC_SERIES in hwmv2, it's done for you by selecting the SOC

File: soc/xlnx/zynq7000/xc7zxxxs/Kconfig.soc (hunk 3 above)
Comment (nordicjm) [reply soc/xlnx/zynq7000/xc7zxxxs/Kconfig.soc:14]: Likewise, you select a SOC and it selects the series for you. And you get a much better representation by looking at the soc.yml file

**Pull request closed without merging**